### DISCUSSION TITLE
Trying to avoid reading the same text twice

### DISCUSSION THREAD

**Discussion by leitvit** [karashiiro/TextToTalk#169] [Ideas]
Hello!

I've been using this extension for a while now with Polly and I love it. 
There are instances, however, when the same text is read twice in a row. I've tried to circumvent this on my own by making it unable to repeat the last read text and it did resolve my issues. 
I don't know if it works as intended, the way I've done it makes the plugin only play the audio the first time when talking to the same NPC multiple times - if the NPC has a single text window.

I see that the issue has been discussed in #136, if my naïve attempt is valid I can open a PR for it or try to do it in a different way. 
[Here is the link to my commit.](https://github.com/karashiiro/TextToTalk/commit/b35c528864a954227f98127f65acef918b92ff9d) 

Feedback is appreciated, thanks!

**Comment by karashiiro** [comment]
Deduplicating text that way doesn't work well from what I've seen, since the sound queue processes messages asynchronously from multiple sources (chat, AddonTalk, and AddonBattleTalk). When any source produces a new message, it overwrites the last played message.

I actually used to do something similar specifically with NPCAnnouncement chat messages, but I removed that in [this commit](https://github.com/karashiiro/TextToTalk/commit/8a243fb0876fced68968eca324225b0a75bf4a2b) since the message sources should be handling that internally. I actually removed a bunch of special message handling at some point since it was hiding several obscure bugs, but that may have accidentally introduced regressions like what you're seeing here.

Do you have any specific reproduction process so I can understand when this happens now? I don't have access to an install to test things myself since I'm out of town, but it'll help for reviewing code.

**Reply by leitvit** [reply]
It doesn't seem to be repeating on a pattern. I'll try and catch a few examples so I can share it!

Thanks for having a look!
